pkg/kernel/networkservice/inject/client.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2020 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inject
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+type injectClient struct{}
+
+// NewClient - returns a new networkservice.NetworkServiceClient that moves given network
+// interface into the Endpoint's pod network namespace on Request and back to Forwarder's
+// network namespace on Close
+func NewClient() networkservice.NetworkServiceClient {
+	return &injectClient{}
+}
+
+func (c *injectClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest,
+	opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	logger := log.FromContext(ctx).WithField("injectClient", "Request")
+	conn, err := next.Client(ctx).Request(ctx, request, opts...)
+	if err != nil {
+		return nil, err
+	}
+	if err := move(logger, conn, false); err != nil {
```

Comment:
We have already done `next.Request()` here and it was successful, so please add here `next.Close()`.
```suggestion
	if err := move(logger, conn, false); err != nil {
		if _, closeErr := next.Client(ctx).Close(ctx, conn, opts...); closeErr != nil {
			logger.Errorf("failed to close failed connection: %s %s", conn.GetId(), closeErr.Error())
		}
```

Reply:
yes! done.